protocols/isis/server/interface.go

```diff
@@ -0,0 +1,299 @@
+package server
+
+import (
+	"bytes"
+	"fmt"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/bio-routing/bio-rd/config"
+	"github.com/bio-routing/bio-rd/protocols/isis/packet"
+	"github.com/bio-routing/bio-rd/protocols/isis/types"
+
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	AllL1ISS  = [6]byte{0x01, 0x80, 0xC2, 0x00, 0x00, 0x14}
+	AllL2ISS  = [6]byte{0x01, 0x80, 0xC2, 0x00, 0x00, 0x15}
+	AllP2PISS = [6]byte{0x09, 0x00, 0x2b, 0x00, 0x00, 0x50}
+	AllISS    = [6]byte{0x09, 0x00, 0x2B, 0x00, 0x00, 0x05}
+	AllESS    = [6]byte{0x09, 0x00, 0x2B, 0x00, 0x00, 0x04}
+)
+
+const (
+	NLPID_IPv4 = uint8(0xcc)
+	NLPID_IPv6 = uint8(0x8e)
+)
+
+type netIf struct {
+	isisServer         *ISISServer
+	name               string
+	ifa                *net.Interface
+	passive            bool
+	p2p                bool
+	l1                 *level
+	l2                 *level
+	socket             int
+	supportedProtocols []uint8
+}
+
+type level struct {
+	HelloInterval uint16
```

Comment:
since level is not exported, the attributes should not be exported too, except they would fulfill a interface (which seems not the case here, looks like copy paste of ISISLevelConfig)